text/setup.py

```diff
@@ -37,6 +37,7 @@
     'nptyping>=1.4.4,<1.5.0',
     'omegaconf>=2.1.1,<2.2.0',
     'sentencepiece>=0.1.95,<0.2.0',
+    'protobuf<=3.18.1',  # https://github.com/awslabs/autogluon/issues/1762
```

Comment:
So there are two versions range of protobuf exists in autogluon now. We probably want to just have 1. Otherwise, which version will be installed is unclear.

Reply:
Agreed, it needs to be specified in the core setup_utils and both text and core must use that method of specifying verison

Reply:
Agreed, it needs to be specified in the core setup_utils and both text and core must use that method of specifying verison

Reply:
Will the dependency resolver in pypi be able to handle that automatically? https://pip.pypa.io/en/stable/topics/dependency-resolution/ . For example, we may not ask core to depend on protobuf /limit the version but rely on the dependency resolver of pypi to figure that out automatically?

Disclaimer: I'm not the expert on that. But constraining the version in core due to version constraint in the sub-package does not sound right to me.

Reply:
Ideally, the version caps in sub-packages such as text/image/timeseries should not impact the version caps in core. This helps these sub-packages be loosely coupled with each other.

Reply:
@sxjscience What you describe makes sense, but it comes at the cost of increased complexity for detecting issues, as now the presence of AutoGluon-Text impacts the version of Protobuf in core, and thus it is harder to test that both versions (with and without text installed) work correctly. I'm open to either option, but would generally recommend having one golden version of package we are confident works across everything than potentially many different versions depending on which subset of modules are installed by the user.

Reply:
My point is that core should not depend on unnecessary packages. For example, users should be able to use core without installing protobuf.

Reply:
So ray also has the protobuf issue, and that's why core might depend on protobuf. Even though the new ray release has the protobuf fix, but it should be addressed in another PR.

Reply:
Ok, I'll leave the details of this to @sxjscience and @yinweisu , I am ok with either approach.

Reply:
If I’m going to revise core, should I revise the version cap in ray @weisy?

Reply:
Actually, I think maybe separate them out and let pypi to resolve the conflict is better because each package might have a different protobuf version range. It would be non-trivial to figure it out manually by ourselves every time. 

In the meanwhile, I'll try to update ray with new release to avoid the protobuf issue on their side